<!-- original : navigation_global/move_base.xml (electric) -->
<launch>
  <!-- source pointcloud -->
  <arg name="cloud_in" default="/multisense/organized_image_points2_color" />
  <!-- base frame to scan -->  
  <arg name="scan_frame" default="pointcloud_to_scan_base" />
  <!-- scan range -->
  <arg name="range_max" default="0.5" /> 
  <arg name="range_min" default="-0.5" />
  <!-- fixed frame -->
  <arg name="scan_height" default="1.0" />
  <arg name="scan_frame_parent" default="BODY" />
  <arg name="use_fixed_frame" default="true" /> <!-- target_frame should be ignored when use_fixed_frame is false -->

  <group if="$(arg use_fixed_frame)" >
    <node pkg="jsk_robot_startup" type="ConstantHeightFramePublisher.py" name="pointcloud_to_laserscan_constant_height_frame" output="screen" >
      <param name="parent_frame" value="$(arg scan_frame_parent)" />
      <param name="height" value="$(arg scan_height)" />
      <param name="frame_name" value="$(arg scan_frame)" />
    </node>
  </group>
  
  <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan_test" output="screen" >
    <remap from="cloud_in" to="$(arg cloud_in)" />
    <remap from="scan" to="pointcloud_scan" />
    <param name="target_frame" value="$(arg scan_frame)" />
    <param name="max_height" value="$(arg range_max)" />
    <param name="min_height" value="$(arg range_min)" />
    <param name="range_min" value="1.5" />    
    <param name="range_max" value="30.0" />
    <param name="use_inf" value="false" />
  </node>
</launch>
